seastar-macros/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+#[proc_macro_attribute]
+pub fn test(
+    args: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let input = syn::parse_macro_input!(item as syn::ItemFn);
+
+    let name = &input.sig.ident;
+    let body = &input.block;
+    let attrs = &input.attrs;
+
+    if input.sig.asyncness.is_none() {
+        let msg = "the async keyword is missing from the function declaration";
+        return syn::Error::new_spanned(input.sig.fn_token, msg)
```

Comment:
What is `fn_token`, what does it span? Function name? signature? whole function?
`input.sig` should be possible to use it, and it sounds like it should span the signature (I may be mistaken of course), which is where the error should be shown.